coverage/sqldata.py

```diff
@@ -181,13 +181,13 @@ def dump(self):                                         # pragma: debugging
     def _file_id(self, filename, add=False):
         """Get the file id for `filename`.
 
-        If filename is not in the database yet, add if it `add` is True.
+        If filename is not in the database yet, add it if `add` is True.
         If `add` is not True, return None.
         """
         if filename not in self._file_map:
             if add:
                 with self._connect() as con:
-                    cur = con.execute("insert into file (path) values (?)", (filename,))
+                    cur = con.execute("insert or replace into file (path) values (?)", (filename,))
```

Comment:
How does this fix relate to #760?

Reply:
You've made the link.. I guess it can help, but can only say it for https://github.com/nedbat/coveragepy/issues/702#issuecomment-432220508 really, although I've not tested the "insert or replace" attempt with that one.